client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
```

Comment:
I'm not sure I understand what you mean.

The create/update methods in an AngularJS controller should do minimal processing.  Ideally they should do this:
```js

function create(invalid) {
  if (invalid) { return; }

  // submit to the server
  service.create(vm.voucher)
  .then(handleSuccess)
  .catch(handleError);
}
```
Of course, sometimes it may be more complex, especially if the page has multiple forms.

However, there shouldn't be a need to for a function call such as `processVoucher()` or `processVoucherItems()`.  If there is any pre-processing to do, is it cleaner to put it in a client-side service.